CONTRIBUTING.txt

```diff
@@ -0,0 +1,99 @@
+FreeCAD Contribution Process (FCP)
+==================================
+
+FreeCAD's contribution process is inspired by the Collective Code Construction Contract which itself is an evolution of the github.com Fork and Pull Model.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+
+0. Status
+
+FreeCAD is in a transition period. The following are to be regarded as GUIDELINES for contribution submission and acceptance. For historical reasons, the actual process MAY diverge from this process during the transition.  Such deviations SHOULD be noted and discussed whenever possible.
+
+1. Goals
+
+    The FreeCAD Contribution Process is expressed here with the following specific goals in mind:
+
+    1. To provide transparency and fairness in the contribution process.
+    2. To allow useful contributions to be included as quickly as possible.
```

Comment:
"meaningful", "useful" and any other version are all very subjective expressions, so should not be enshrined in rules at all. Clearly identify exactly **who** will decide and this issue vanishes. At the end of the day it will always come down to a human/person/bot? to decide.

Reply:
> Clearly identify exactly **who** will decide and this issue vanishes.

This is very limited. Probably it's about more generally determining **what are the criteria** to decide usefulness/meaningfulness. But that's another topic, and probably one of our next step.

Reply:
Please express the concern you have.  What bad thing could happen if we leave the text as is?  What better thing could happen if we change it to 'meaningful'?

Reply:
No clear things here. Probably nothing bad will happen. Just 'useful' calls more interrogations to me : to who ? in which extent ? 'Meaningful' brings more clarity : does my patch makes sense for at least some people ?

Reply:
18 is a sub rule of "specific goals" and cannot be read in isolation. 
So in that light the actual goal is simple:  "encourage contributions to be included as quickly as possible". The goal is clear and unburdened with responsibilities.